src/rshim_usb.c

```diff
@@ -140,8 +140,13 @@ static struct rshim_usb_addr get_wvalue_windex(int chan, int addr, uint16_t ver_
   struct rshim_usb_addr rsh_usb_addr;
 
   if (ver_id == RSHIM_BLUEFIELD_3) {
-    rsh_usb_addr.wvalue = bf3_wvalue_widx_pair_map[chan].wvalue + BF_MMIO_BASE;
-    rsh_usb_addr.windex = bf3_wvalue_widx_pair_map[chan].windex + addr;
+    if (chan <= 0xF) {
+      rsh_usb_addr.wvalue = bf3_wvalue_widx_pair_map[chan].wvalue + BF_MMIO_BASE;
+      rsh_usb_addr.windex = bf3_wvalue_widx_pair_map[chan].windex + addr;
```

Comment:
I don't understand the logic here.  CRspace addresses are 28 bits.  Isn't wvalue the upper 16 bits of the address?  In that case it should never be > 0xF.  Also I don't think there's any need to add in BF_MMIO_BASE since that just sets bit 28.